src/server/pkg/deploy/assets/assets.go

```diff
@@ -579,6 +579,47 @@ func EtcdStatefulSet(opts *AssetOpts, diskSpace int) interface{} {
 		etcdCmd[i] = fmt.Sprintf("\"%s\"", str) // quote all arguments, for shell
 	}
 
+	var pvcTemplates []interface{}
+	switch backend {
+	case googleBackend, amazonBackend:
+		pvcTemplates = []interface{}{
+			map[string]interface{}{
+				"metadata": map[string]interface{}{
+					"name":   etcdVolumeClaimName,
+					"labels": labels(etcdName),
+					"annotations": map[string]string{
+						"volume.beta.kubernetes.io/storage-class": etcdStorageClassName,
+					},
+				},
+				"spec": map[string]interface{}{
+					"resources": map[string]interface{}{
+						"requests": map[string]interface{}{
+							"storage": resource.MustParse(fmt.Sprintf("%vGi", diskSpace)),
+						},
+					},
+					"accessModes": []string{"ReadWriteOnce"},
+				},
+			},
+		}
+	default:
+		pvcTemplates = []interface{}{
```

Comment:
Yeah will factor that out.

The idea is that you can use `storageclass` if you are on GCE or AWS, because that's where dynamically provisioning is officially supported.  If you are on Azure, you can still specify a PVC template and dynamic provisioning will work, according to my own testing.